### PR TITLE
feat(coinbase): initiate coinbase pay flow

### PR DESCRIPTION
**Pull request by awtsui**
### Description

When the coinbase pay card is pressed, the coinbase pay flow begins. In addition, the coinbase pay card shows if, but not only, CELO is chosen. This PR adds functionality to the coinbase pay card, which lies behind a feature flag.

https://user-images.githubusercontent.com/57046916/177885898-609d3bcc-fa11-4154-b11e-2cff352f0240.mp4

Once user logs into Coinbase account, they are shown this screen to authorize transactions with Valora.

### Other changes

N/A

### Tested

Tested coinbase pay card's visibility depending on selected asset – only CELO is allowed at the moment.
Clicking through coinbase pay experience, selected asset and amount was successfully shown on "Send Celo" screen. 

### How others should test

Testing use of connected bank account and completion of a real transaction has not been done yet.
- Have not ran Valora on mainnet 

### Related issues

- Fixes #[345]
- https://linear.app/valora/issue/ACT-345/initiate-coinbase-pay-flow

### Backwards compatibility

N/A

### CODE REVIEW

**Comment by MuckT** [comment]
@awtsui does it make sense to put this behind a feature flag so we can remotely disable it if issues arise?

File: src/fiatExchanges/CoinbasePaymentSection.tsx [hunk 1]
```diff
@@ -1,29 +1,63 @@
...
 
-  if (!coinbaseProvider || coinbaseProvider.restricted || !coinbasePayEnabled) {
+  const appIdResponse = useAsync(async () => readOnceFromFirebase('coinbasePay/appId'), [])
+  const appId = appIdResponse.result
+
+  if (
+    !coinbaseProvider ||
+    coinbaseProvider.restricted ||
+    !coinbasePayEnabled ||
+    !allowedAssets.includes(digitalAsset)
```
Comment (jh2oman) [inline src/fiatExchanges/CoinbasePaymentSection.tsx:42]: I think you should add `!appId` to this list. If for some reason we can't fetch appId from firebase, we probably don't want to display coinbase as an option

Comment (awtsui) [reply src/fiatExchanges/CoinbasePaymentSection.tsx:43]: addressed!

File: src/fiatExchanges/CoinbasePaymentSection.tsx [hunk 2]
```diff
@@ -1,29 +1,64 @@
...
-export function CoinbasePaymentSection({ coinbaseProvider }: CoinbasePaymentSectionProps) {
+export function CoinbasePaymentSection({
+  digitalAsset,
+  cryptoAmount,
+  coinbaseProvider,
+}: CoinbasePaymentSectionProps) {
   const { t } = useTranslation()
   const coinbasePayEnabled = useSelector(coinbasePayEnabledSelector)
+  const walletAddress = useSelector(walletAddressSelector)!
+  const allowedAssets = [CiCoCurrency.CELO]
```
Comment (jophish) [inline src/fiatExchanges/CoinbasePaymentSection.tsx:33]: super nit: we check if the `digitalAsset` is one from a list of `allowedAssets`, but when generating the onramp URL, we hardcode it to `CGLD` (corresponding to `CiCoCurrency.CELO`). Can we just similarly hardcode a check that `digitalAsset === CiCoCurrency.CELO`? Once/if Coinbase allows multiple different assets, we'll have to determine which asset to give to the flow by using a map from `digitalAsset`, at which point a list for `allowedAssets` seems more appropriate.

**Review by jophish — COMMENTED** [review]
This looks great! :slightly_smiling_face: just pending tests passing

File: src/fiatExchanges/CoinbasePaymentSection.tsx (hunk 2 above)
Comment (awtsui) [reply src/fiatExchanges/CoinbasePaymentSection.tsx:33]: Fair enough. It was a half way modular attempt.

**Review by jh2oman — APPROVED** [review]
Looks great!

**Review by jophish — APPROVED** [review]

**Comment by awtsui** [comment]
Leaving a comment here for testing documentation purposes.

Full flow test on mainnet:
 ~ Specific, detailed instructions will be written on an alfajores testing plan.
 ~ Enabled coinbase pay feature flag locally

Navigate to "Add & Withdraw" in side menu
Navigate to "Add Funds"
Select "CELO" and an amount, I did 4
Coinbase Pay card should show up. Upon clicking it, I was navigated to a webview, where I logged to their Coinbase account, verify, and authorize transactions with Valora. In California, only crypto transfer is allowed, so I bought some CELO on Coinbase for the test. The rest is pretty straight forward. Select "crypto balance" and carry out transaction. The end of the flow does seem a little clunky as I had to backtrack every step of the flow to return to the main screen.

**Comment by awtsui** [comment]
Coinbase Pay Experience - Alfajores Testing Documentation

Cash in/ Transfer from Coinbase with Coinbase Pay

-----Steps-----
1. Navigate to side menu and tap Add & Withdraw button
2. Tap on Add Funds button
3. Tap on CELO button to select CELO
4. Enter any amount of CELO, and tap the Next button
5. Tap on the Coinbase Pay card 
6. If this is you're first time using Coinbase Pay, login to Coinbase account
7. Enter 2-step verification code
8. Tap on Authorize button
9. Tap Your balance button and select a payment method
10. Tap left arrow button on the top left
11. Tap Preview send
12. Tap Confirm button

-----Expected Results-----
1. ~
2. ~
3. Coinbase Pay currently only supports the transfer of Celo, so user must select CELO
4. ~
5. Tapping Coinbase Pay card opens Coinbase Pay flow
Steps 6 - 8 are only necessary if user has not logged in, verified, and authorized their Coinbase account
6. Entering login information brings user to 2-step verification page
7. Completing 2-step verification brings user to authentication page
8. Tapping Authorize brings user to Send Celo screen with Preview send button
9. Tapping Your balance button should reveal a menu of personal payment methods
10. Tapping left arrow button brings user back to Send Celo screen
11. Tapping Preview send bring user to Order preview screen with confirm button
12. Tapping Confirm should initiate transaction and user should see whether or not the transaction was successful

For testing on Alfajores, I did not complete the Coinbase Pay flow as funds may be lost, or difficult to retrieve. The entire flow was, however, completed on a locally enabled mainnet build, where Celo was successfully transferred from a Coinbase account to a Valora wallet

**Comment by awtsui** [comment]
> Coinbase Pay Experience - Alfajores Testing Documentation
> 
> Cash in/ Transfer from Coinbase with Coinbase Pay
> 
> -----Steps-----
> 
> 1. Navigate to side menu and tap Add & Withdraw button
> 2. Tap on Add Funds button
> 3. Tap on CELO button to select CELO
> 4. Enter any amount of CELO, and tap the Next button
> 5. Tap on the Coinbase Pay card
> 6. If this is you're first time using Coinbase Pay, login to Coinbase account
> 7. Enter 2-step verification code
> 8. Tap on Authorize button
> 9. Tap Your balance button and select a payment method
> 10. Tap left arrow button on the top left
> 11. Tap Preview send
> 12. Tap Confirm button
> 
> -----Expected Results-----
> 
> 1. ~
> 2. ~
> 3. Coinbase Pay currently only supports the transfer of Celo, so user must select CELO
> 4. ~
> 5. Tapping Coinbase Pay card opens Coinbase Pay flow
>    Steps 6 - 8 are only necessary if user has not logged in, verified, and authorized their Coinbase account
> 6. Entering login information brings user to 2-step verification page
> 7. Completing 2-step verification brings user to authentication page
> 8. Tapping Authorize brings user to Send Celo screen with Preview send button
> 9. Tapping Your balance button should reveal a menu of personal payment methods
> 10. Tapping left arrow button brings user back to Send Celo screen
> 11. Tapping Preview send bring user to Order preview screen with confirm button
> 12. Tapping Confirm should initiate transaction and user should see whether or not the transaction was successful
> 
> For testing on Alfajores, I did not complete the Coinbase Pay flow as funds may be lost, or difficult to retrieve. The entire flow was, however, completed on a locally enabled mainnet build, where Celo was successfully transferred from a Coinbase account to a Valora wallet

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above task and observed that

**Comment by ValoraQA** [comment]
Hey @MuckT we have verified the above task on latest iOS Alfajores build and latest Android Internal alfajores build V:1.37.0 

Observation : 
- We are not able to see Coinbase Pay when we tried for US number
- We are not able to see coinbase Pay as we tried to use the app with US VPN 
- Even if we are entering 100$ or more amount as seen in the video still we are not able to see the coinbase pay 

Attachment : [coinbase.mp4](https://drive.google.com/file/d/16aWbMK4SA8DGR3-vUelY4FGQb0i1l4Od/view?usp=sharing)

Devices : Google Pixel 2XL(11.0) , Google Pixel 4a(12.0), iPhone 12(14.7.1)

**Comment by MuckT** [comment]
@awtsui @jh2oman @jophish based on the findings for #2724 a user can get stuck trying to link the same bank account. Is there a way to avoid attempting relink the account if it's already populated?

**Pull request merged**